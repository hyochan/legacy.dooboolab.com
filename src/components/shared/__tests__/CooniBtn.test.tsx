import React from 'react';
import { shallow } from 'enzyme';

import { CooniBtn } from '../CooniBtn';

// const component = shallow(
//     <CooniBtn white={true} btnTxt='CooniBtn 1st test' />
//   );

// test for the pure component
describe('CooniBtn shared component test', () => {
  const component = shallow(
    <CooniBtn white={true} btnTxt='CooniBtn 2nd test' />,
  );

  it('component and snapshot matches', () => {
    expect(component).toMatchSnapshot();
  });

  it('component onClick function has to be added and work to make the count 2', () => {
    let count = 1;
    const onClick = () => {
      count++;
    };

    component.setProps({ onClick: () => onClick() });
    expect(component).toMatchSnapshot();

    component.find('div').simulate('click');
    expect(count).toBe(2);
  });
});
