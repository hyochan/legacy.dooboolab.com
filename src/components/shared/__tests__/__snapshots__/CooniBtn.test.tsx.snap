// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CooniBtn shared component test component and snapshot matches 1`] = `ShallowWrapper {}`;

exports[`CooniBtn shared component test component onClick function has to be added and work to make the count 2 1`] = `ShallowWrapper {}`;
